apps/dashboard/app/api/auth/[...nextauth]/route.ts

```diff
@@ -1,57 +1,64 @@
+import NextAuth, { AuthOptions } from "next-auth";
 import { fetchUserData } from "@/services/graphql/queries/me-data";
-import NextAuth from "next-auth"
+import { env } from "@/env";
+import { ApolloQueryResult } from "@apollo/client";
+import { MeQuery } from "@/services/graphql/generated";
 
-const type = "oauth" as const // as ProviderType
+declare module "next-auth" {
+  interface Profile {
+    id: string;
+  }
+  interface Session {
+    sub: string | null;
+    accessToken: string;
+    userData: ApolloQueryResult<MeQuery>;
+  }
+}
 
-export const authOptions = {
-  // Configure one or more authentication providers
+const type = "oauth" as const;
+export const authOptions: AuthOptions = {
   providers: [
     {
       id: "blink",
-      clientId: process.env.CLIENT_ID,
-      clientSecret: process.env.CLIENT_SECRET,
-      wellKnown: "http://127.0.0.1:4444/.well-known/openid-configuration",
-      useSecureCookies: false,
+      clientId: env.CLIENT_ID,
+      clientSecret: env.CLIENT_SECRET,
+      wellKnown: env.HYDRA_PUBLIC,
       authorization: {
         params: { scope: "offline transactions:read payments:send" },
       },
       idToken: false,
       name: "Blink",
       type,
       profile(profile) {
-        console.log({ profile }, "profile123");
         return {
           id: profile.sub,
-          // email: profile.email,
         };
       },
     },
-    // ...add more providers here
   ],
   debug: true,
-  secret: process.env.NEXTAUTH_SECRET as string,
+  secret: env.NEXTAUTH_SECRET,
   callbacks: {
     async jwt({ token, account, profile }) {
       // Persist the OAuth access_token and or the user id to the token right after signin
       if (account) {
         token.accessToken = account.access_token;
         token.expiresAt = account.expires_at;
         token.refreshToken = account.refresh_token;
-        token.id = profile.id;
+        token.id = profile?.id;
       }
       return token;
     },
     async session({ session, token, user }) {
-      const userData = await fetchUserData(token.accessToken); 
-      // Send properties to the client, like an access_token from a provider.
-      session.sub = token.sub;
-      session.accessToken = token.accessToken;
+      const userData = await fetchUserData(token.accessToken as string);
+      session.sub = token?.sub || null;
```

Comment:
I don't think null is an acceptable fallback. I think we want to return an error in that scenario